src/main/java/amazon/DealsOfTodayPage.java

```diff
@@ -60,7 +60,9 @@ public boolean itemSelect() {
     }
 
     public void waitForPageLoad() {
-        new WebDriverWait(driver, Duration.ofSeconds(20)).until(ExpectedConditions.
-                elementToBeClickable(listOfItemsInPage.get(listOfItemsInPage.size()-1)));
+        if(listOfItemsInPage.size()>1){
```

Comment:
And what if size isn't greater than 1 ? Did you think about what will happen if the condition is false and how much it is logical ?
Once I've said already. You must wait till the elements amount is more than 0!! Fix the method.